app/models/ems_refresh/save_inventory_physical_infra.rb

```diff
@@ -58,4 +59,35 @@ def save_asset_details_inventory(parent, hash)
     return if hash.nil?
     save_inventory_single(:asset_details, parent, hash)
   end
+
+  def ipaddress?(hostname)
+    IPAddr.new(hostname)
+    return true
+  rescue
+    return false
+  end
+
+  def resolve_hostname(ipaddress, ems)
+    ems.hostname = "https://#{Resolv.getname(ipaddress)}/"
+    _log.info("EMS ID: #{ems.id}" + " Resolved hostname successfully.")
+  rescue => err
+    _log.warn("EMS ID: #{ems.id}" + " It's not possible resolve hostname of the physical infra, #{err}.")
+  end
+
+  def resolve_ip_address(hostname, ems)
+    ems.ipaddress = Resolv.getaddress(hostname)
```

Comment:
@agrare you had a PR recently that changed the refresh process to treat the EMS as a first class item in inventory parsing, right?

I think that would work better here than doing more parsing in the `save_inventory` logic.